### ISSUE TITLE
NS4000 false positive when using an initializer

### ISSUE THREAD

**Issue by darrenbiel** [nsubstitute/NSubstitute.Analyzers#124]
This seems similar to issue #47 but I am still seeing this in 1.0.10.

When using an initializer in a returns, I get an analysis error NS4000 even if there is no nested returns.

```csharp
public class TestClass
{
    [Fact]
    public void NS4000_FalsePositive()
    {
        IBook book = SetupBook();
        var bookFinder = Substitute.For<IBookFinder>();

        bookFinder.FavoriteBook.Returns(book); // Doesn't shows analysis error
        bookFinder.Books.Returns(new List<IBook>() { book }); // Shows analysis error
        bookFinder.BookPromoter.Returns(new BookPromoter() { PromotedBook = book }); // Shows analysis error

        Assert.Contains(book, bookFinder.Books);
    }

    private static IBook SetupBook()
    {
        var model = Substitute.For<IBook>();
        model.IsFiction.Returns(true);
        return model;
    }
}

public interface IBook
{
    bool IsFiction { get; set; }
}

public interface IBookFinder
{
    ICollection<IBook> Books { get; }
    IBook FavoriteBook { get; }
    IBookPromoter BookPromoter { get; }
}

public interface IBookPromoter
{
    IBook PromotedBook { get; set; }
}

public class BookPromoter : IBookPromoter
{
    public IBook PromotedBook { get; set; }
}
```

**Comment by tpodolak** [comment]
Thanks for reporting, your issue has the same root-cause as #106 which was fixed by #107. Fix will be available in next version
